apps/src/code-studio/components/aiTutor/aiTutorPanel.jsx

```diff
@@ -1,29 +1,93 @@
-import React from 'react';
+import React, {useState} from 'react';
 import PropTypes from 'prop-types';
+import {useDispatch} from 'react-redux';
 import i18n from '@cdo/locale';
 import style from './ai-tutor.module.scss';
+import Checkbox from '@cdo/apps/componentLibrary/checkbox';
 import AITutorPanelContainer from '@cdo/apps/code-studio/components/aiTutor/aiTutorPanelContainer';
-import CompilationAssistant from './compilationAssistant';
+import CompilationTutor from './compilationTutor';
+import ValidationTutor from './validationTutor';
+import GeneralChatTutor from './generalChatTutor';
+import {addAIResponse} from '@cdo/apps/aiTutor/redux/aiTutorRedux';
 const icon = require('@cdo/static/ai-bot.png');
 
-export default class AITutorPanel extends React.Component {
-  static propTypes = {
-    level: PropTypes.shape({
-      id: PropTypes.number.isRequired,
-      type: PropTypes.string.isRequired,
-    }),
-  };
+const AITutorPanel = ({level}) => {
+  const dispatch = useDispatch();
+  const isCodingLevel = level.type === 'Javalab';
+
+  const initialCheckboxes = [
+    {
+      key: 'compilation',
+      label: 'Code compilation',
+      checked: false,
+      hidden: !isCodingLevel,
+    },
+    {
+      key: 'validation',
+      label: 'Failing tests',
+      checked: false,
+      hidden: !level.hasValidation,
+    },
+    {
+      key: 'question',
+      label: 'I have a question',
+      checked: false,
+      hidden: false,
+    },
+  ];
 
-  render() {
-    const {level} = this.props;
-    const isCodingLevel = level.type === 'Javalab';
+  const [checkboxes, updateCheckboxes] = useState(initialCheckboxes);
 
-    return (
-      <AITutorPanelContainer level={level}>
-        <h3 id="ai_tutor_panel">AI Tutor</h3>
-        <img alt={i18n.aiBot()} src={icon} className={style.aiBotImg} />
-        {isCodingLevel && <CompilationAssistant levelId={level.id} />}
-      </AITutorPanelContainer>
+  const handleCheckboxChange = key => {
+    dispatch(addAIResponse(''));
+    const updatedCheckboxes = checkboxes.map(checkbox =>
+      checkbox.key === key
+        ? {...checkbox, checked: !checkbox.checked}
+        : {...checkbox, checked: false}
     );
-  }
-}
+    updateCheckboxes(updatedCheckboxes);
+  };
+
+  const getChecked = checkboxes => {
+    return checkboxes.find(checkbox => checkbox.checked);
+  };
+
+  const checked = getChecked(checkboxes);
+  const compilationSelected = checked && checked.key === 'compilation';
+  const validationSelected = checked && checked.key === 'validation';
+  const questionSelected = checked && checked.key === 'question';
+
+  return (
+    <AITutorPanelContainer level={level}>
+      <h3 id="ai_tutor_panel">AI Tutor</h3>
+      <img alt={i18n.aiBot()} src={icon} className={style.aiBotImg} />
+      <div>
+        <h4> What would you like AI Tutor to help you with?</h4>
+        {checkboxes.map(
+          cb =>
+            !cb.hidden && (
+              <Checkbox
```

Comment:
if we do radio buttons instead of checkboxes that will enforce that only once can be selected right?

Reply:
@molly-moen @ebeastlake Updated to Radio Buttons! 

https://github.com/code-dot-org/code-dot-org/assets/12300669/8aeada57-bd3e-4c57-8162-e4eeda0e83ad